ikarus/finiteelements/mechanics/materials/hyperelastic/volumetric/interface.hh

```diff
@@ -0,0 +1,97 @@
+// SPDX-FileCopyrightText: 2021-2025 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file interface.hh
+ * \brief Implementation of the volumetric part of a hyperelastic material.
+ * \ingroup  materials
+ */
+
+#pragma once
+
+#include <ikarus/finiteelements/mechanics/materials/hyperelastic/concepts.hh>
+#include <ikarus/finiteelements/physicshelper.hh>
+#include <ikarus/utils/tensorutils.hh>
+
+namespace Ikarus::Materials {
+
+/**
+ * \brief Interface for the volumetric part of a hyperelastic material. Has to be parametrized with a volumetric
+ * function.
+ * \ingroup materials
+ *
+ * \details The volumetric part of the hyperelastic model, i.e., related to \f$ U(J) \f$, is
+ * parametrized with a certain volumetric function (VF) implemented in terms of the determinant of the deformation
+ * gradient. The underlying volumetric function must only implement the energy \f$ U(J) \f$ and its first
+ * and second derivatives w.r.t \f$ J \f$.
+ *
+ * \tparam VF Volumetric function.
+ */
+template <Concepts::VolumetricFunction VF>
+struct Volumetric
+{
+  using VolumetricFunction = VF;
+  using MaterialParameter  = double;
+
+  [[nodiscard]] constexpr static std::string name() noexcept { return "Volumetric function: " + VF::name(); }
+
+  /**
+   * \brief Construct a new volumetric part
+   *
+   * \param matPar Materialparameter that is considered as the penalty parameter for the constraint against volumetric
+   * deformations. Typically, if the deviatoric part of the energy function is a function of the total principal
+   * stretches, then this material parameter is the  Lamé's first parameter and if the energy is a function of the
+   * deviatoric principal stretches, then bulk modulus is used.
+   *
+   * \param vf the volumetric function
+   */
+  Volumetric(MaterialParameter matPar, const VF vf)
```

Comment:
The VF function should be a forwarding reference. and not t const value. You have to template the constructor and create a deduction guide unfortunately. see newton or similar